src/main/java/com/zendesk/maxwell/SynchronousBootstrapper.java

```diff
@@ -0,0 +1,200 @@
+			row.getData("completed_at") != null &&
+			( long ) row.getData("is_complete") == 1;
+	}
+
+	@Override
+	public boolean isBootstrapRow(RowMap row) {
+		return row.getDatabase().equals("maxwell") &&
+			row.getTable().equals("bootstrap");
+	}
+
+	@Override
+	public boolean shouldSkip(RowMap row) {
+		// the synchronous bootstrapper blocks other incoming messages
+		// to the replication stream so there's nothing to skip
+		return false;
+	}
+
+	@Override
+	public void startBootstrap(RowMap startBootstrapRow, Schema schema, AbstractProducer producer, OpenReplicator replicator) throws Exception {
+		String databaseName = ( String ) startBootstrapRow.getData("database_name");
+		String tableName = ( String ) startBootstrapRow.getData("table_name");
+		LOGGER.debug(String.format("bootstrapping request for %s.%s", databaseName, tableName));
+		Database database = findDatabase(schema, databaseName);
+		Table table = findTable(tableName, database);
+		BinlogPosition position = new BinlogPosition(replicator.getBinlogPosition(), replicator.getBinlogFileName());
+		producer.push(startBootstrapRow);
+		producer.push(replicationStreamBootstrapStartRow(table, position));
+		LOGGER.info(String.format("bootstrapping started for %s.%s, binlog position is %s", databaseName, tableName, position.toString()));
+		try ( Connection connection = getConnection() ) {
+			setBootstrapRowToStarted(startBootstrapRow, connection);
+			ResultSet resultSet = getAllRows(databaseName, tableName, connection);
+			int insertedRows = 0;
+			while ( resultSet.next() ) {
+				RowMap row = new RowMap(
+						"insert",
+						databaseName,
+						tableName,
+						System.currentTimeMillis(),
+						table.getPKList(),
+						position);
+				setRowValues(row, resultSet, table);
+				LOGGER.debug("bootstrapping row : " + row.toJSON());
+				producer.push(row);
+				++insertedRows;
+				updateInsertedRowsColumn(insertedRows, startBootstrapRow, connection);
+			}
+			setBootstrapRowToCompleted(insertedRows, startBootstrapRow, connection);
+		}
+	}
+
+	private void updateInsertedRowsColumn(int insertedRows, RowMap startBootstrapRow, Connection connection) throws SQLException {
+		long now = System.currentTimeMillis();
+		if ( now - lastInsertedRowsUpdateTimeMillis > INSERTED_ROWS_UPDATE_PERIOD_MILLIS ) {
+			long rowId = ( long ) startBootstrapRow.getData("id");
+			String sql = "update maxwell.bootstrap set inserted_rows = ? where id = ?";
+			PreparedStatement preparedStatement = connection.prepareStatement(sql);
+			preparedStatement.setInt(1, insertedRows);
+			preparedStatement.setLong(2, rowId);
+			preparedStatement.execute();
+			lastInsertedRowsUpdateTimeMillis = now;
+		}
+	}
+
+	protected Connection getConnection( ) throws SQLException {
+		return context.getConnectionPool().getConnection();
+	}
+
+	private RowMap replicationStreamBootstrapStartRow(Table table, BinlogPosition position) {
+		return replicationStreamBootstrapRow("bootstrap-start", table, position);
```

Comment:
ok, so the `bootstrap-start` and `bootstrap-complete` events *do* have a db/table name.  That's not documented in bootstrap.md, can you fix?